README.md

```diff
@@ -1,69 +1,89 @@
-​
++ `GetSecretValue` on the secrets you want to retrieve.
++ `ListSecrets` on all secrets.
++ \(Optional\) `Decrypt` on the KMS key if the secrets are encrypted with a customer managed key.
+
+For more information, see [Authentication and access control for AWS Secrets Manager](auth-and-access.md).
+
 ### Usage
-​
+
 To use the action, add a step to your workflow that uses the following syntax.
-​
+
 ```
 - name: Step name
   uses: aws-actions/aws-secretsmanager-get-secrets@v1
   with:
     secret-ids: |
       secretId1
-      ENV_VAR, secretId2
+      ENV_VAR_NAME, secretId2
     parse-json-secrets: (Optional) true|false
 ```
+Parameters
 
-### Parameters
-- `secret-ids`: Secret ARNS, names, and name prefixes. 
+- `secret-ids` Secret ARNS, names, and name prefixes. 
 
-By default, the step creates each environment variable name from the secret name, transformed to include only uppercase letters, numbers, and underscores, and so that it doesn't begin with a number. 
+By default, the step creates each environment variable name from the secret name, transformed to include only uppercase letters, numbers, and underscores, and so that it doesn't begin with a number.   
 
-To set the environment variable name, enter it before the secret ID, followed by a comma. For example `ENV_VAR_1, secretId` creates an environment variable named **ENV_VAR_1** from the secret `secretId`. 
+To set the environment variable name, enter it before the secret ID, followed by a comma. For example `ENV_VAR_1, secretId` creates an environment variable named **ENV\_VAR\_1** from the secret `secretId`. 
 
-The environment variable name can consist of uppercase letters, numbers, and underscores.  
+The environment variable name can consist of uppercase letters, numbers, and underscores. 
 
 To use a prefix, enter at least three characters followed by an asterisk. For example `dev*` matches all secrets with a name beginning in **dev**. The maximum number of matching secrets that can be retrieved is 100. If you set the variable name, and the prefix matches multiple secrets, then the action fails.
-​
+
 - `parse-json-secrets`
 
 (Optional - default false) By default, the action sets the environment variable value to the entire JSON string in the secret value. 
 
 Set `parse-json-secrets` to `true` to create environment variables for each key/value pair in the JSON.
 
 Note that if the JSON uses case-sensitive keys such as "name" and "Name", the action will have duplicate name conflicts. In this case, set `parse-json-secrets` to `false` and parse the JSON secret value separately. 
-​
+
+### Environment variable naming
+
+The environment variables created by the action are named the same as the secrets they comes from. If you parse the JSON of the secret, then the environment variable name includes both the secret name and the JSON key name, for example `MYSECRET_KEYNAME`. 
```

Comment:
Underscore ('_') is a valid character in both secret names and JSON keys. As a result, {secret: FOO_SECRET, key: KEY} and {secret: FOO, key: SECRET_KEY} would result in the same environment variable name (FOO_SECRET_KEY) and undefined behavior. Looking at the code it looks like, the last entry in the for loop would win. I know this is not new behavior, but since you now have this "Environment variable naming" section, could we actually add a note describing this behavior? Essentially define the undefined behavior.

Reply:
done!